addons/library.xbmc.addon/libXBMC_addon.h

```diff
@@ -47,6 +47,8 @@ typedef intptr_t      ssize_t;
 #define ADDON_HELPER_ARCH       "powerpc-osx"
 #elif defined(__arm__)
 #define ADDON_HELPER_ARCH       "arm-osx"
+#elif defined(__aarch64__)
+#define ADDON_HELPER_ARCH       "arm-osx"
```

Comment:
it's a little strange, yes. even stranger is arm-osx to begin with. arm-darwin makes more sense. darwin is the base, then you have darwin/osx, darwin/iOS, darwin/tvOS plus 32 and 64 bit flavors. 

This is all dependent on naming of the actual libs and they must match. 

In reality, one will never have the different arch libs in the sample place so they do not really need different naming.
In practice, it does not matter what naming they have as long as core and libs match.